lib.ml

```diff
@@ -18,13 +18,19 @@ let alias_env : Types.tycon_environment =
 
 let datatype = DesugarDatatypes.read ~aliases:alias_env
 
-let cgi_parameters = ref []
-let cookies = ref []
-(*  http_response_headers: this is state for the webif interface. I hope we can
-    find a better way for library functions to communicate with the web
-    interface. *)
-let http_response_headers = ref []
-let http_response_code = ref 200
+type requestData = {
+  cgi_parameters : (string * string) list ref;
+  cookies : (string * string) list ref;
+  http_response_headers : (string * string) list ref;
+  http_response_code : int ref
+}
+
+let empty_request_data () = {
```

Comment:
Nitpick: This generates a new (empty) request_data, rather than defining a unique empty structure, so maybe call it "new_request_data"?